code/game/gamemodes/events/holidays/Holidays.dm

```diff
@@ -42,6 +42,8 @@ var/global/Holiday = null
 					current_holidays += ST_PATRICKS_DAY
 
 		if(4) // Apr
+			if(DD != 1 && time2text(world.timeofday, "DDD") == "Sun") //no idea what would happen if april fools gets set twice so we check just in case
```

Comment:
Ok, so. Each round, it looks over the list of current_holidays and chooses one at random. Adding the same entry multiple times would increase the chances of it being picked. For example, if April 22nd (Earth Day) was a Sunday, both EARTH_DAY and APRIL_FOOLS_DAY would be added to the list, and it would randomly pick between them. Having APRIL_FOOLS_DAY on the list twice would simply increase the odds of it being picked (on the currently impossible case of April 1st having holidays besides APRIL_FOOLS_DAY). You can simply keep it as the day check here.

And to confirm, yes, it can only be one Holiday at once. Can't be both APRIL_FOOLS_DAY and ST_PATRICKS_DAY at the same time.